### PR TITLE
Updating type annotations in models and traits

### PR DESCRIPTION
**Pull request by ILDaviz**
Hi, I refactored the model and trait section. Because I can't continue with the graphql module development if I don't fix the types in the existing models. I updated all the models present and in some traits where I needed.
Let me know if this is okay.

Best regards and congratulations on the project.

### CODE REVIEW

File: packages/core/src/Models/Product.php [hunk 1]
```diff
@@ -79,50 +82,40 @@ protected static function newFactory(): ProductFactory
 
     /**
      * Returns the attributes to be stored against this model.
-     *
-     * @return array
      */
-    public function mappedAttributes()
+    public function mappedAttributes(): array
```
Comment (alecritson) [inline packages/core/src/Models/Product.php:86]: I think the docblock was incorrect here, as it's referencing a relationship I believe it should be `Collection` that is returned.

Comment (ILDaviz) [reply packages/core/src/Models/Product.php]: Wooops I missed it.

**Review by alecritson — APPROVED** [review]
Thanks for this @ILDaviz ! :)

**Comment by ILDaviz** [comment]
If you need more help typing the rest of the project just ask. 🚀

**Comment by sanneterpstra** [comment]
When using the mappedAttributes method I now get an error:

`Lunar\Models\Product::mappedAttributes(): Return value must be of type Lunar\Models\Collection, Illuminate\Database\Eloquent\Collection returned`

**Pull request merged**